packages/browser/src/eventbuilder.ts

```diff
@@ -149,8 +158,17 @@ export function eventFromException(
 ): PromiseLike<Event> {
   const syntheticException = (hint && hint.syntheticException) || undefined;
   const event = eventFromUnknownInput(stackParser, exception, syntheticException, attachStacktrace);
-  addExceptionMechanism(event); // defaults to { type: 'generic', handled: true }
+  const providedMechanism: Mechanism | undefined =
+    hint && hint.data && (hint.data as { mechanism: Mechanism }).mechanism;
+  const mechanism: Mechanism = providedMechanism || {
+    handled: true,
+    type: 'generic',
+  };
+
+  addExceptionMechanism(event, mechanism);
```

Comment:
Can we change the way of passing on the mechanism to using an eventprocessor on the scope? Similarly to how we do it in [remix here](https://github.com/getsentry/sentry-javascript/blob/195fe33df1e42383cc05a1a58b607d9b8728bf15/packages/remix/src/utils/instrumentServer.ts#L73-L85)? If this is possible, we'd prefer it method over adding it to the hint.